notebooks/Transformation.ipynb

```diff
@@ -0,0 +1,297 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# necessery libraries\n",
+    "import pandas as pd\n",
+    "import sys\n",
+    "from sklearn.preprocessing import StandardScaler\n",
+    "sys.path.insert(0,'../src/utils/')\n",
+    "import transform_data as TRD"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# set option setting to display all columns\n",
+    "pd.set_option('display.max_columns', None)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#load data\n",
+    "data = pd.read_csv(\"../data/featsel/Citywide_Survey_featsel.csv\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# drop unneeded column ('Unnamed: 0')\n",
+    "data.drop(columns=['Unnamed: 0'], inplace=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# show data\n",
+    "data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# list date and categorical features\n",
+    "not_numeric_features = list(data.dtypes[data.dtypes == object].index)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(not_numeric_features)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "date_features = ['first_travel_date','last_travel_date']  #date features \n",
+    "categorical_features = ['cms_zone', 'work_cms_zone']  #categorical features\n",
+    "\n",
+    "# all features to be encoded include categorical and labeled features\n",
+    "features_to_encode = ['cms_zone', 'work_cms_zone','smartphone_type','made_trips', 'employment',\n",
+    "'age', 'student','vehicle',  'congestion','drive_cbd_freq', 'planning_apps','job_type', 'industry',\n",
+    "'telework_freq', 'work_mode','bike_freq', 'tnc_freq', 'tnc_purpose',\n",
+    "'tnc_mode', 'packages', 'harassment', 'gender', 'education', 'english_proficiency', 'ethnicity']"
+   ]
+  },
```

Comment:
yes we can ,but we must reverse the order in for these features:
tnc_fre, telework_freq, planning_apps, drive_cbd_freq, congestion, english_proficiency

Reply:
We could use mapping to maintain the required order of data

Reply:
+ to reverse the needed values

Reply:
we have 3 types of missing 
1-3?
or 0-2?
then 3/4 to categories-1 ?

Reply:
what do you think is better?

Reply:
I think we do not have to do reverse label encoding since the algorithm depends on the distance the arrange reversed or not it is not important

and the numbers are arranged, so we don't need to do label encoding either, we just want to do the mapping to replace missing codes with other values in the same arrange.